File: packages/backend3/src/main/kotlin/quizzem/backend/features/question/api/dto/QuestionDto.kt
package quizzem.backend.features.question.api.dto

import quizzem.backend.core.api.dto.QuizzemDto
import quizzem.backend.features.question.model.AnswerMode
import quizzem.backend.features.question.model.MediaType
import java.time.Instant
import java.util.*

class QuestionDto(
    id: UUID,
    createdAt: Instant,
    updatedAt: Instant,
    var text: String,
    var answerMode: AnswerMode,
    var mediaType: MediaType,
    var answers: List<AnswerDto>
) : QuizzemDto(id, createdAt, updatedAt)

File: packages/backend/src/features/question/api/dto/question.dto.ts
import { QuizzemDto } from 'src/core/api/quizzem.dto';
import { AnswerDto } from 'src/features/question/api/dto/answer.dto';
import { AnswerMode } from 'src/features/question/model/answer-mode';
import { MediaType } from 'src/features/question/model/media-type';

export class QuestionDto extends QuizzemDto {
  text!: string;
  answerMode!: AnswerMode;
  mediaType!: MediaType;
  answers!: AnswerDto[];
}
